static/js/popovers.js

```diff
@@ -1050,6 +1198,80 @@ exports.register_click_handlers = function () {
             last_scroll = date;
         });
     }());
+
+    $("body").on("click", "#send_later", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        popovers.toggle_send_later_popover(this);
+    });
+
+    $("body").on("click", ".send_later_hours", function (e) {
+        var send_later_in = $(e.currentTarget).attr('id');
+        var send_at_time = moment().add(send_later_hours[send_later_in].hours, 'h');
+        var message_content = compose_state.message_content();
+        var command = "/schedule " + send_at_time.format("MMM D YYYY h:mm a") + "\n";
+
+        if (reminder.is_deferred_delivery(message_content)) {
+            var raw_message = message_content.split('\n');
+            message_content = raw_message.slice(1).join('\n');
+        }
+
+        message_content = command + message_content;
+        compose_state.message_content(message_content);
```

Comment:
For the actual compose box manipulation, it would be better to use `compose_ui.insert_syntax_and_focus`; it's a function we use for this sort of thing that does a much better job of dealing with existing compose box content.  

That said, I don't think the right long-term mechanism for this widget is to add `/schedule` in the compose box; I think before we enable this outside the development environment, we will want to instead keep track of this state in a separate variable `compose_state.send_at_time` (which is null for non-send-later messages), that state should adjust the "Send" button text to say e.g. "Schedule" and we'd check that state in `compose.finish` to determine whether to send or schedule the outgoing message.  

I think the right strategy is to do the refactor to move this to `compose_actions.js` and use `insert_syntax_and_focus` now, get this merged, and then we can fix the above in a follow-up commit (there's several other unrelated things we'll want to fix, too).

Reply:
@timabbott, the problem with using `insert_syntax_and_focus` is, it inserts the given syntax at the current position of the caret. Instead of that, we always want to insert the `/schedule` syntax at the beginning of the message content. Is it possible to insert at the beginning even after cursor being at some other position using `insert_syntax_and_focus`? 
I think to do this, we might need some additional refactoring in the code, where we optionally provide a parameter about where to insert the syntax.
A side note, I'm back to work on my WIP PRs from today onwards. Sorry for much delay.

Reply:
Let's just change this to not use the slash command at all, as I suggested in my last comment.  I think conceptually a slash command should be a way to schedule messages, but if you're going to click a button, we can do a much more native-looking UI by tracking the data via `compose_state.send_at_time` and showing the planning timestamp near the "Send" button.

Reply:
Alright, now I get it. So we just want to change the state of the "Send" button by knowing information from `compose_state.send_at_time`. If we have some valid value in that variable, we'll show Send later type of button, otherwise(if that variable is None), we would show normal send button.

Reply:
Yep, and we can display the target send time in some nice output to the left of the button (maybe where the "press enter to send" text usually lives)